doc/final-report/galois-final-report.tex

```diff
@@ -0,0 +1,291 @@
+    \item A \href{https://github.com/GaloisInc/ckzg-eip-4844-verification/tree/main/spec/Spec/BlsEC}{Cryptol specification}
+          of the elliptic curve operations required by the Python KZG Specification.
+          We modeled the \href{https://github.com/ethereum/py_ecc}{py\_ecc} implementation
+    \item A \href{https://github.com/GaloisInc/ckzg-eip-4844-verification/blob/main/spec/Spec/BlsSerde.cry}{Cryptol specification}
+          for the serialization and deserialization of EC Points used by the Python KZG Specification.
+          We modeled the \href{https://github.com/ethereum/py_ecc}{py\_ecc} implementation
+    \item A \href{https://github.com/GaloisInc/ckzg-eip-4844-verification/tree/main/spec/TestVectors}{test bench}
+          from randomized inputs to show correctness of the Cryptol specification and the Python KZG Specification
+          \begin{itemize}
+                \item The test inputs were generated using the \texttt{:dumptests} REPL command in Cryptol.
+                This is Cryptol's built-in randomized, fuzz testing generator. We added these tests into our Cryptol source
+                \item We also created a Git patch file that adds these tests to the Python KZG Specification test bench
+          \end{itemize}
+    \item A \href{https://github.com/GaloisInc/ckzg-eip-4844-verification/tree/main/proof}{SAW proof script}
+          that proves a couple of the \href{https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md\#bit-reversal-permutation}
+          {"bit-reversal permutation" functions} in the Cryptol specification equivalent to the c-kzg functions
+    \item An additional, \href{https://github.com/GaloisInc/ckzg-eip-4844-verification/tree/main/spec/LLVM}{partial Cryptol specification}
+          for some of the functions in 1 written to match the Python KZG Specificatiothat did not complete
+          formal verification with SAW in 5
+\end{enumerate}
+
+There is a README in the repository for each of these. These READMEs include details on how to run all the tests,
+randomized test vectors, and proofs for repeatable results.
+
+\subsection{Cryptol Implementation of the Python KZG Specification}
+
+Table \ref{table:python-to-cryptol} gives a mapping of each function in the Python KZG Specification
+to the corresponding Cryptol function for reference.
+
+
+\begin{table}
+
+\textbf{
+    \caption{Python KZG Specification mapping to Cryptol}
+    \label{table:python-to-cryptol}
+}
+\begin{center}
+    \begin{tabular}{ |c|c|c| }
+        \hline
+        \textbf{Category} & \textbf{Python KZG Specification} & \textbf{Cryptol} \\
+        \hline
+        \multirow{3}{10em}{\href{https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md\#bit-reversal-permutation}{Bit-reversal permutation}}
+            & \href{https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md\#is_power_of_two}{is\_power\_of\_two}
```

Comment:
Don't know if it matters much but these are all currently tracking head rather than a specific commit and therefore could easily break in the future.

Reply:
I agree. Done.